app/src/components/features/mocksV2/MockEditorIndex/Editor/HeaderSection/index.tsx

```diff
@@ -0,0 +1,115 @@
+import { useState } from "react";
+import { DeleteOutlined } from "@ant-design/icons";
+import { AutoComplete, Button, Col, Input, Row } from "antd";
+import HEADER_SUGGESTIONS from "config/constants/sub/header-suggestions";
+import { ValidationErrors } from "../../types";
+import { useMediaQuery } from "react-responsive";
+
+interface HeaderSectionProps {
+  mappedHeader: {
+    name: string;
+    value: string;
+    index: number;
+  }[];
+  errors: {
+    headers: { indexError: number; valueError: string }[];
+  };
+  updateHeaders: (value: string, index: number, type: string) => void;
+  removeHeader: (index: number) => void;
+  addHeader: () => void;
+}
+
+export default function HeaderSection({
+  mappedHeader,
+  errors,
+  updateHeaders,
+  removeHeader,
+  addHeader,
+}: HeaderSectionProps) {
+  const [filteredOptions, setFilteredOptions] = useState(HEADER_SUGGESTIONS.Response);
+  const isSmallScreen = !useMediaQuery({ query: "(max-width: 768px)" });
+
+  const renderHeaderErrors = (errors: ValidationErrors, index: number, errorType: string) => {
+    if (errors.headers && errors.headers.some((err) => err.indexError === index)) {
+      return (
+        <div key={index}>
+          {errors.headers
+            .filter((err) => err.indexError === index)
+            .map((headerError, errorIndex) => {
+              if (headerError.valueError.startsWith(errorType)) {
+                const boxPosError = headerError.valueError.slice(errorType.length).trim();
+                return <div key={errorIndex}>{boxPosError}</div>;
+              }
+              return null;
+            })}
+        </div>
+      );
+    }
+    return null;
+  };
```

Comment:
We should not perform checks on strings to determine if the error is for header `name` or `value`, error messages might change in future which will break this.